src/ops/key.rs

```diff
@@ -115,57 +113,69 @@ pub fn save_keys(
         }
     }
 
-    match diesel::insert_into(keys::table)
-        .values(keys_to_insert)
-        .execute(&mut db::establish_connection()?)
-    {
-        Ok(num_inserted_rows) => {
-            assert_eq!(num_rows_to_insert, num_inserted_rows);
-            Ok(())
-        }
-        Err(e) => {
-            error!("Failed to save records to db: {:?}", e);
-            Err(Error::internal_server_error())
-        }
+    let pool = &db::establish_connection().await?;
+    let mut num_inserted_rows: u64 = 0;
+    for key in keys_to_insert{
+        let result = match sqlx::query!(
+            r#"INSERT INTO keys (id, master_sae_id, slave_sae_id, size, content)VALUES ($1, $2, $3, $4, $5);"#,
```

Comment:
*Suggestion:* Here and in similar places, if it adds any value, consider using the sqlx macro `sqlx::query_file_as!`. This allows for segregating the SQL statements into dedicated files, taking them out of Rust code.

Reply:
Good idea. Thanks for following-up, resolving this thread.